drivers/i2c/i2c_ll_stm32_v1_rtio.c

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <soc.h>
+#include <stm32_ll_i2c.h>
+#include <stm32_ll_rcc.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c/rtio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/kernel.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/sys/util.h>
+
+#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(i2c_ll_stm32_v1_rtio);
+
+#include "i2c_ll_stm32.h"
+#include "i2c-priv.h"
+
+#define I2C_REQUEST_WRITE       0x00
+#define I2C_REQUEST_READ        0x01
+#define HEADER                  0xF0
+
+static void i2c_stm32_disable_transfer_interrupts(const struct device *dev)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	I2C_TypeDef *i2c = cfg->i2c;
+
+	LL_I2C_DisableIT_TX(i2c);
+	LL_I2C_DisableIT_RX(i2c);
+	LL_I2C_DisableIT_EVT(i2c);
+	LL_I2C_DisableIT_BUF(i2c);
+}
+
+static void i2c_stm32_enable_transfer_interrupts(const struct device *dev)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	I2C_TypeDef *i2c = cfg->i2c;
+
+	LL_I2C_EnableIT_ERR(i2c);
+	LL_I2C_EnableIT_EVT(i2c);
+	LL_I2C_EnableIT_BUF(i2c);
+}
+
+static void i2c_stm32_generate_start_condition(I2C_TypeDef *i2c)
+{
+	uint16_t cr1 = LL_I2C_ReadReg(i2c, CR1);
+
+	if (cr1 & I2C_CR1_STOP) {
+		LOG_DBG("%s: START while STOP active!", __func__);
+		LL_I2C_WriteReg(i2c, CR1, cr1 & ~I2C_CR1_STOP);
+	}
+
+	LL_I2C_GenerateStartCondition(i2c);
+}
+
+static void i2c_stm32_master_mode_end(const struct device *dev)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	struct i2c_stm32_data *data = dev->data;
+	struct i2c_rtio *ctx = data->ctx;
+	I2C_TypeDef *i2c = cfg->i2c;
+
+	i2c_stm32_disable_transfer_interrupts(dev);
+	LL_I2C_Disable(i2c);
+	if (data->xfer_len == 0 && i2c_rtio_complete(ctx, 0)) {
+		i2c_stm32_start(dev);
+	}
+
+}
+
+static void handle_sb(const struct device *dev)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	struct i2c_stm32_data *data = dev->data;
+	I2C_TypeDef *i2c = cfg->i2c;
+
+	uint16_t saddr = data->slave_address;
+	uint8_t slave;
+
+	if (data->xfer_flags & I2C_MSG_ADDR_10_BITS) {
```

Comment:
`if (data->xfer_flags & I2C_MSG_ADDR_10_BITS != 0U)`

Equivalent comments on a few other tests implementation (line 102, 120, 129, 163, 175, ...)

Ditto in **drivers/i2c/i2c_ll_stm32_v2_rtio.c**.

Reply:
Fixed